### PR TITLE
[heartbleeder.go] using flags

### PR DESCRIPTION
**Pull request by matiasinsaurralde**
Using flags. It's possible to set timeout, etc.

### CODE REVIEW

File: heartbleeder.go [hunk 1]
```diff
@@ -11,11 +12,19 @@ import (
 )
 
 func main() {
-	if len(os.Args) != 2 {
-		fmt.Printf("Usage: %s host[:443]\n", os.Args[0])
-		os.Exit(2)
+        var timeout int
+        flag.IntVar(&timeout, "timeout", 10, "Timeout after sending heartbeat")
```
Comment (titanous) [inline heartbleeder.go]: Make it a `flag.Duration`.

File: heartbleeder.go [hunk 2]
```diff
@@ -11,11 +12,19 @@ import (
...
+
+	flag.Usage = func() {
+		fmt.Printf("Usage: %s [options] [host][:443]\n", os.Args[0])
```
Comment (titanous) [inline heartbleeder.go]: `host` shouldn't be in brackets.

File: heartbleeder.go [hunk 3]
```diff
@@ -11,11 +12,19 @@ import (
...
+		fmt.Println("Options:")
+		flag.PrintDefaults()
 	}
-	host := os.Args[1]
+
+	flag.Parse()
+
+	host := flag.Args()[0]
```
Comment (titanous) [inline heartbleeder.go]: `flag.Arg(0)`

**Comment by matiasinsaurralde** [comment]
Updated (Y)

File: heartbleeder.go [hunk 4]
```diff
@@ -11,11 +12,18 @@ import (
...
+		fmt.Println("Options:")
+		flag.PrintDefaults()
 	}
-	host := os.Args[1]
+
+	flag.Parse()
+
+	host := flag.Arg(0)
+
 	if !strings.Contains(host, ":") {
```
Comment (titanous) [inline heartbleeder.go:23]: `gofmt` and drop each of the empty lines above this one.

**Comment by titanous** [comment]
Thanks! Just clean up the whitespace, squash into one commit, and I'll merge.

**Comment by matiasinsaurralde** [comment]
No problem, thanks for the tool!

**Comment by titanous** [comment]
@matiasinsaurralde LGTM, can you squash?

**Comment by matiasinsaurralde** [comment]
Did I something wrong? It seems that this PR doesn't update after squash

**Comment by titanous** [comment]
Did you force push to the branch? You can always open a new PR if things are wacky.

**Pull request closed without merging**